### PR TITLE
Added script to populate maven repos during the build for the openshift-middleware image 

### PR DESCRIPTION
**Pull request by tqvarnst**

### CODE REVIEW

**Comment by tqvarnst** [comment]
@BenHall Would be great if you can take a look at the script I've created and if you see something that is wrong, please let me know. Otherwise, I will merge this pull request tomorrow.

**Comment by BenHall** [comment]
The script itself looks good! Thank you!

File: environments/openshift-middleware/scripts/middleware.sh [hunk 1]
```diff
@@ -33,3 +33,5 @@ docker pull bucharestgold/centos7-s2i-nodejs
 # docker pull registry.access.redhat.com/jboss-fuse-6/fis-karaf-openshift
 
 curl -k -L -o /opt/jboss-image-streams.json https://raw.githubusercontent.com/redhat-middleware-hackathon/openshift-files/master/jboss-image-streams.json
+
+sh populate-maven-repos.sh
```
Comment (BenHall) [inline environments/openshift-middleware/scripts/middleware.sh]: Having this in the `packer.json` could make it easier to see when scripts are called and help debugging when things fail.

https://github.com/openshift-evangelists/intro-katacoda/blob/master/environments/openshift-middleware/openshift-middleware.json#L35-L36

Comment (tqvarnst) [reply environments/openshift-middleware/scripts/middleware.sh]: Ahh, didn't now that we could do that. I agree that would be much better.

**Comment by BenHall** [comment]
build output:

```
    openshift-middleware: - ADDING A MAVEN SETTINGS FILE
    openshift-middleware: - CREATING A TEMPORARY POM FILE TO PRE-POPULATE PLUGINS
    openshift-middleware: - BUILDING THE TEMPORARY POM
    openshift-middleware: Nov 23, 2017 11:08:38 AM org.apache.maven.wagon.providers.http.httpclient.impl.execchain.RetryExec execute
    openshift-middleware: INFO: I/O exception (org.apache.maven.wagon.providers.http.httpclient.NoHttpResponseException) caught when processing request to {s}->https://maven.repository.redhat.com:443: The target server failed to respond
    openshift-middleware: Nov 23, 2017 11:08:38 AM org.apache.maven.wagon.providers.http.httpclient.impl.execchain.RetryExec execute
    openshift-middleware: INFO: Retrying request to {s}->https://maven.repository.redhat.com:443
    openshift-middleware: Nov 23, 2017 11:08:38 AM org.apache.maven.wagon.providers.http.httpclient.impl.execchain.RetryExec execute
    openshift-middleware: INFO: I/O exception (org.apache.maven.wagon.providers.http.httpclient.NoHttpResponseException) caught when processing request to {s}->https://maven.repository.redhat.com:443: The target server failed to respond
    openshift-middleware: Nov 23, 2017 11:08:38 AM org.apache.maven.wagon.providers.http.httpclient.impl.execchain.RetryExec execute
    openshift-middleware: INFO: Retrying request to {s}->https://maven.repository.redhat.com:443
    openshift-middleware: Nov 23, 2017 11:08:38 AM org.apache.maven.wagon.providers.http.httpclient.impl.execchain.RetryExec execute
    openshift-middleware: INFO: I/O exception (org.apache.maven.wagon.providers.http.httpclient.NoHttpResponseException) caught when processing request to {s}->https://maven.repository.redhat.com:443: The target server failed to respond
    openshift-middleware: Nov 23, 2017 11:08:38 AM org.apache.maven.wagon.providers.http.httpclient.impl.execchain.RetryExec execute
    openshift-middleware: INFO: Retrying request to {s}->https://maven.repository.redhat.com:443
    openshift-middleware: - DELETE THE TEMPORARY POM
    openshift-middleware: - CLONING REPOSITORIES
    openshift-middleware: -- CLONING REPO https://github.com/openshift-katacoda/rhoar-getting-started.git
    openshift-middleware: - ITERATING OVER PROJECT AND BUILDING THEM
    openshift-middleware: -- BUILDING PROJECT ./rhoar-getting-started/wildfly-swarm
    openshift-middleware: Resolving 330 out of 392 artifacts
    openshift-middleware: Resolving 63 out of 392 artifacts
    openshift-middleware: -- BUILDING PROJECT ./rhoar-getting-started/vertx
    openshift-middleware: - DONE
```

**Comment by tqvarnst** [comment]
Hi Ben,
I wanted to log the output of the commands and the echo to a file so that
we could A. Check which version of the image we where using, B. See any
potential build errors.

Are you saying that I shouldn’t do that?

I could redirect each command to the log file?

Cheers
Thomas


tors 23 nov. 2017 kl 12:39 skrev Ben Hall <notifications@github.com>:

**Comment by BenHall** [comment]
Can we pipe to both stdout and a file? Without pipeline to stdout it make
debugging builds impossible.

**Comment by tqvarnst** [comment]
My new pull request uses tee which pipes both to stdout and a file. Please
review and merge if you think it looks correct.

fre 24 nov. 2017 kl 15:48 skrev Ben Hall <notifications@github.com>:

**Comment by tqvarnst** [comment]
https://github.com/openshift-evangelists/intro-katacoda/pull/80


fre 24 nov. 2017 kl 16:53 skrev Thomas Qvarnstrom <tqvarnst@redhat.com>:

**Pull request merged**